packages/openapi-v3/src/__tests__/integration/controller-spec.integration.ts

```diff
@@ -203,6 +207,131 @@ describe('controller spec', () => {
     });
   });
 
+  it('allows operations to provide definition of referenced models through #/components/schema', () => {
+    class MyController {
+      @get('/todos', {
+        responses: {
+          '200': {
+            description: 'Array of Category model instances',
+            content: {
+              'application/json': {
+                schema: {
+                  $ref: '#/components/schemas/Todo',
+                  definitions: {
+                    Todo: {
+                      title: 'Todo',
+                      properties: {
+                        title: {type: 'string'},
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      })
+      async find(): Promise<object[]> {
+        return []; // dummy implementation, it's never called
+      }
+    }
+
+    const spec = getControllerSpec(MyController);
+    const opSpec: OperationObject = spec.paths['/todos'].get;
+    const responseSpec = opSpec.responses['200'].content['application/json'];
+    expect(responseSpec.schema).to.deepEqual({
+      $ref: '#/components/schemas/Todo',
+    });
+
+    const globalSchemas = (spec.components || {}).schemas;
+    expect(globalSchemas).to.deepEqual({
+      Todo: {
+        title: 'Todo',
+        properties: {
+          title: {
+            type: 'string',
+          },
+        },
+      },
+    });
+  });
+
+  it('allows operations to provide definition of referenced models through #/definitions', () => {
+    class MyController {
+      @get('/todos', {
+        responses: {
+          '200': {
+            description: 'Array of Category model instances',
+            content: {
+              'application/json': {
+                schema: {
+                  $ref: '#/definitions/Todo',
+                  definitions: {
+                    Todo: {
+                      title: 'Todo',
+                      properties: {
+                        title: {type: 'string'},
+                      },
+                    },
+                  },
+                },
+              },
+            },
+          },
+        },
+      })
+      async find(): Promise<object[]> {
+        return []; // dummy implementation, it's never called
+      }
+    }
+
+    const spec = getControllerSpec(MyController);
+    const opSpec: OperationObject = spec.paths['/todos'].get;
+    const responseSpec = opSpec.responses['200'].content['application/json'];
+    expect(responseSpec.schema).to.deepEqual({
+      $ref: '#/definitions/Todo',
+    });
+
+    const globalSchemas = (spec.components || {}).schemas;
+    expect(globalSchemas).to.deepEqual({
+      Todo: {
+        title: 'Todo',
+        properties: {
+          title: {
+            type: 'string',
+          },
+        },
+      },
+    });
+  });
+
+  it('returns an empty object when it cannot find definition of referenced model', () => {
```

Comment:
Should we report it as an error for unresolved `$ref` or defer it to other definitions outside of the controller?